cpp/src/search/search.cu

```diff
@@ -100,21 +101,29 @@ std::unique_ptr<column> search_ordered(table_view const& t,
 
   // This utility will ensure all corresponding dictionary columns have matching keys.
   // It will return any new dictionary columns created as well as updated table_views.
-  auto matched  = dictionary::detail::match_dictionaries({t, values}, stream);
-  auto d_t      = table_device_view::create(matched.second.front(), stream);
-  auto d_values = table_device_view::create(matched.second.back(), stream);
+  auto matched = dictionary::detail::match_dictionaries({t, values}, stream);
```

Comment:
If the struct columns can contain dictionary columns you may want to do the dictionary match after the flatten.
@hyperbolic2346

Reply:
@jrhemstad talked me out of the possibility of dictionaries inside structs and the join code will not match after the flatten.